### PR TITLE
Start documenting a set of API design principles

### PR DESCRIPTION
**Pull request by dhellmann**
We've had several design discussions recently in which I have needed
to reiterate these principles, so I think it's time to write them
down in one place.

### CODE REVIEW

File: design/api-design-principles.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+provisional
+
+## Table of Contents
+
+A table of contents is helpful for quickly jumping to sections of a
+design and for highlighting any additional information provided beyond
+the standard template.
+
+[Tools for generating][] a table of contents from markdown are available.
```
Comment (russellb) [inline design/api-design-principles.md]: want to remove this boilerplate?

**Comment by russellb** [comment]
I agree with what you've written so far.-

File: design/api-design-principles.md (hunk 1 above)
Comment (dhellmann) [reply design/api-design-principles.md]: Done

**Comment by dhellmann** [comment]
> I like this so far.
> 
> Another one we could add is: don't make assumptions about what the BaremetalHost will be used for. The Machine API is not the only consumer and running a Kubernetes node is not the only thing users may want to do with the Host.

I like that. Do you want me to add it to this, or should we consider this list something we maintain as we go along?

**Comment by zaneb** [comment]
I'm happy either way. If you want to merge now I'm happy to submit another PR for it, or you can update this PR.

**Comment by dhellmann** [comment]
Let's do it separately. It's a good way to reinforce that the list is meant to be updated over time.

**Comment by dhellmann** [comment]
See #75 for the follow-up.

**Pull request merged**